index.html

```diff
@@ -101,19 +199,623 @@ <h3>
               The port that the socket should be bound to. When omitted, or 0,
               any port can be used.
             </dd>
+            <div class="note">
+              The only current implementation (Chromium) ignores
+              {{TCPSocketOptions/localAddress}} and {{TCPSocketOptions/localPort}}.
+            </div>
+            <dt>
+              <dfn>noDelay</dfn> member
+            </dt>
+            <dd>
+              Enables the `TCP_NODELAY` option,
+              disabling <a href=
+              "https://en.wikipedia.org/wiki/Nagle%27s_algorithm">Nagle's
+              algorithm</a>.
+            </dd>
+            <dt>
+              <dfn>keepAlive</dfn> member
+            </dt>
+            <dd>
+              Enables/disables TCP Keep-Alive by setting `SO_KEEPALIVE` option on the socket. 
+            </dd>
+            <dt>
+              <dfn>keepAliveDelay</dfn> member
+            </dt>
+            <dd>
+              Specifies the Keep-Alive ping delay in milliseconds. Must be specified only if
+              {{TCPSocketOptions/keepAlive}} is set to `true`.
+            </dd>
+          </dl>
+
+        </section>
+
+      </section>
+
+      <section>
+        <h3>{{TCPSocket/[[readable]]}} attribute (internal)</h3>
+
+        <aside class="example">
+          An application receiving data from a TCP socket will typically use the {{ReadableStream}} like this:
+
+          <pre class="js">
+            const { readable } = await socket.connection;
+
+            const reader = readable.getReader();
+
+            while (true) {
+              const { value, done } = await reader.read();
+              if (done) {
+                // |reader| has been canceled.
+                break;
+              }
+              // Do something with |value|...
+            }
+
+            reader.releaseLock();
+          </pre>
+        </aside>
+      
+        The steps to <dfn data-lt="initialize tcp readable stream">initialize the TCP readable stream</dfn> are:
+
+        <ol>
+          <li>Let |stream:ReadableStream| be a [=new=] {{ReadableStream}}.
+          <li>
+            Let |pullAlgorithm| be the following steps [=in parallel=]:
+            <ol>
+              <li>
+                Invoke the operating system to read a tcp packet from the socket, placing the result in the [=byte sequence=] |bytes|.
```

Comment:
You're not reading individual TCP packets. Like in the Web Serial API you want to see what "desiredSize" is on [[readable]] and read that many bytes.

Reply:
Hm. Are you sure this is the right way to go? I believe that our case is different from serial since there we attempt to read actual data from the device, and here we exchange messages with a remote (and I would expect them to come as full packets, not just random bytes).

Reply:
They come as full packets but the OS doesn't give them to us that way. It has an internal buffer which gets filled by incoming packets and when we read we pull out the bytes that have accumulated. That buffer filling up is what applies backpressure to the sender.